Server/Admin/routers/admin_reward_analytics.py

```
@@ -57,9 +75,50 @@ def _get_cache_key(self, key: str) -> str:
         """Get cache key for analytics data"""
         return f"analytics:rewards:{key}"
 
+    def _ensure_redis_connection(self, redis_client, redis_name: str):
+        """Ensure Redis connection is alive, reconnect if needed"""
+        if not redis_client:
+            return False
+        try:
+            redis_client.ping()
+            return True
```

Ruff (0.14.3)

Consider moving this statement to an `else` block

(TRY300)